python/revng/pypeline/utils/__init__.py

```diff
@@ -2,14 +2,96 @@
 # This file is distributed under the MIT License. See LICENSE.md for details.
 #
 
+import asyncio
+import base64
+import hashlib
 import os
+from collections.abc import Buffer
+from contextlib import asynccontextmanager
 from pathlib import Path
+from typing import AsyncIterator, Generic, TypeVar
 
 from xdg import xdg_cache_home
 
+T = TypeVar("T")
+
 
 def cache_directory() -> Path:
     if "PYPELINE_CACHE_DIR" in os.environ:
         return Path(os.environ["PYPELINE_CACHE_DIR"])
     else:
         return xdg_cache_home() / "pypeline"
+
+
+def is_mime_type_text(mime_type: str) -> bool:
+    """
+    Returns True if the mime type represents a text file, False otherwise.
+    """
+    return mime_type.startswith("text/") or mime_type in {
+        "application/json",
+        "application/xml",
+        "application/x-yaml",
+    }
+
+
+def bytes_to_string(data: Buffer, is_text: bool = False) -> str:
+    """Standard way to convert bytes to string in the codebase."""
+    if is_text:
+        return bytes(data).decode("utf-8")
+    return base64.b64encode(data).decode("utf-8")
+
+
+def string_to_bytes(data: str, is_text: bool = False) -> bytes:
```

Comment:
Drop the default `= False`, this is an important argument, it should be explicitly specified in all call sites